server/src/main/java/org/elasticsearch/cluster/health/ClusterStateHealth.java

```diff
@@ -144,6 +149,40 @@ public ClusterStateHealth(
         this.indices = indices;
     }
 
+    public static ClusterHealthStatus getHealthStatus(
```

Comment:
We could move this to `AllocationService` couldn't we? Then we won't have to pass the logger around.